### PR TITLE
fix cross files not being passed to scan-build

### LINKED ISSUE #5492: ninja scan-build does not take --cross-file into account
I am using Meson to cross compile from Linux to a different [(non-mainstream) OS](https://github.com/managarm/managarm). `clang` is used as a compiler. This works fine, but `ninja scan-build` does not.

Output:
```
ninja -v scan-build
[0/1] /usr/local/bin/meson --internal scanbuild /home/korona/Projects/bootstrap-managarm/src/managarm /home/korona/Projects/bootstrap-managarm/build/pkg-builds/managarm-system /usr/local/bin/meson -Dbuild_drivers=true -Dbuild_kernel=false -Dbuild_tools=false
scan-build: Using '/home/korona/Projects/bootstrap-managarm/build/tools/host-llvm-toolchain/bin/clang-8' for static analysis
The Meson build system
Version: 0.50.1
Source dir: /home/korona/Projects/bootstrap-managarm/src/managarm
Build dir: /home/korona/Projects/bootstrap-managarm/build/pkg-builds/managarm-system/meson-private/tmpwe0du8ks
Build type: native build
Project name: managarm
Project version: undefined
Native C compiler: /home/korona/Projects/bootstrap-managarm/build/tools/host-llvm-toolchain/bin/../libexec/ccc-analyzer (gcc 6.3.0 "gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516")
Native C++ compiler: /home/korona/Projects/bootstrap-managarm/build/tools/host-llvm-toolchain/bin/../libexec/c++-analyzer (gcc 6.3.0 "g++ (Debian 6.3.0-18+deb9u1) 6.3.0 20170516")
Build machine cpu family: x86_64
Build machine cpu: x86_64
```

Expected output:
- `meson` should configure its temporary build directory using the same cross file as the directory that `ninja scan-build` is invoked from. The cross file is required to pass arguments such as `-target` and `-gcc-toolchain` to `clang`.
- `meson` should realize that it is doing a `Build type: cross build` and not a `Build type: native build`.

Output during original configuration (correct):
```
The Meson build system
Version: 0.50.1
Source dir: /home/korona/Projects/bootstrap-managarm/src/managarm
Build dir: /home/korona/Projects/bootstrap-managarm/build/pkg-builds/managarm-system
Build type: cross build
Project name: managarm
Project version: undefined
Native C compiler: cc (gcc 6.3.0 "cc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516")
Cross C compiler: clang (clang 8.0.0)
Native C++ compiler: c++ (gcc 6.3.0 "c++ (Debian 6.3.0-18+deb9u1) 6.3.0 20170516")
Cross C++ compiler: clang++ (clang 8.0.0)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Build machine cpu family: x86_64
Build machine cpu: x86_64
```

### PR DESCRIPTION
**Pull request by Apache-HB**
fixes #5492

### CODE REVIEW

**Comment by avdgrinten** [comment]
Can confirm that #5492 is fixed after applying this patch.

File: mesonbuild/coredata.py [hunk 1]
```diff
@@ -743,7 +743,7 @@ def set_options(self, options, *, subproject='', warn_unknown=True):
         if unknown_options and warn_unknown:
             unknown_options = ', '.join(sorted(unknown_options))
             sub = 'In subproject {}: '.format(subproject) if subproject else ''
-            mlog.warning('{}Unknown options: "{}"'.format(sub, unknown_options))
+            mlog.warning('{}Compiler does not support: "{}"'.format(sub, unknown_options))
```
Comment (dcbaker) [inline mesonbuild/coredata.py:746]: This is unrelated.

File: mesonbuild/coredata.py [hunk 2]
```diff
@@ -946,7 +946,7 @@ def write_cmd_line_file(build_dir, options):
 
     properties = OrderedDict()
     if options.cross_file:
-        properties['cross_file'] = options.cross_file
+        properties['cross_file'] = list(map(os.path.abspath, options.cross_file))
```
Comment (dcbaker) [inline mesonbuild/coredata.py:949]: I have two concerns, first is why this isn't being done for native files as well? the second is what happens when you pass a file that isn't an absolute path, like those that are in ~/.local/share/meson/cross.

I think what you really need to do is reuse the code in mesonbuild.coredata.Coredata.__load_config_files

Comment (dcbaker) [reply mesonbuild/coredata.py:949]: Actually, the more I think abut it, all this needs to be is
```python
if options.cross_file:
    properties['cross_files'] = options.cross_file
if options.native_file:
    properties['native_files'] = options.config_file
```

File: mesonbuild/scripts/scanbuild.py [hunk 3]
```diff
@@ -37,6 +38,13 @@ def run(args):
     logdir = os.path.join(blddir, 'meson-logs/scanbuild')
     shutil.rmtree(logdir, ignore_errors=True)
 
+    cmd = coredata.get_cmd_line_file(blddir)
+    data = coredata.CmdLineFileParser()
+    data.read(cmd)
+
+    if 'cross_file' in data['properties'].keys():
+        meson_cmd.append('--cross-file={}'.format(literal_eval(data['properties']['cross_file'])[0]))
+
```
Comment (dcbaker) [inline mesonbuild/scripts/scanbuild.py:47]: This is wrong too, this should be something like:
```python
if 'cross_file' in data['properties']:  # note: not .keys()
    meson_cmd.extend(['-cross-file={}'.format(f) for f in literal_eval(data['properties']['cross_file']))])
```

And the same for native files. You need to load all of the cross (and native files), not just the first one.

**Comment by jpakkane** [comment]
Ping. This also now conflicts and needs to get the tests fixed.
